mpc_node:
  ros__parameters:
    debug: False
    # Publish pose path for visualizaion in RViz
    pub_pose_path: True

    # Sampling time in seconds
    # IMPORTANT!!! Make sure this is the same as the reference trajectory sampling time
    dt_pred: 0.1

    # Number of MPC horizon steps, shoud be > 1
    mpc_window: 5

    # Minimum altitude above ground in UAV local frame, meter(s)
    # This is just for safety, to avoid crashing into ground
    minimum_altitude: -1.0

    # Localization frame id. Pose trajectory is published  with respect to it
    reference_frame_id: "interceptor/odom"

    # Smooth control signal, to avoid large control jumps 
    enable_control_smoothing: False
    # Weight/penality on control input differences
    smooth_input_weight: 100.0

    # 3D velocity limits in m/s. Should be >= 0
    # [x,y,z]
    max_velocity: [10.0, 10.0, 10.0]

    # 3D Net acceleration limits in m/s/s. Should be >= 0
    # [x,y,z]
    max_acceleration: [5.0, 5.0, 5.0]

    # 3D jerk limits in m/s/s/s. Should be >= 0
    # Not used in 6dof implementation; can be used in 9dof when input is jerk
    # [x,y,z]
    max_jerk: [2.0, 2.0, 2.0]

    # Optimization penalty on the states (Q)
    state_weight: 7000.0 # 7000

    # Optimization penalty on the control inuputs (R)
    input_weight: 500.0 # 0.5, 500